packages/backend/src/ee/services/ScimService/ScimService.ts

```diff
@@ -1375,6 +1374,107 @@ export class ScimService extends BaseService {
         }
     }
 
+    private convertLightdashRoleToScimRole(
+        role: Role,
+        type: ScimRoleType,
+    ): ScimRole {
+        const id = role.roleUuid;
+        return {
+            schemas: [ScimSchemaType.ROLE],
+            id,
+            value: id,
+            display: role.name,
+            type,
+            supported: true,
+            meta: {
+                resourceType: 'Role',
+                created: role.createdAt || undefined,
+                lastModified: role.updatedAt || undefined,
+                location: new URL(
+                    `/api/v1/scim/v2/Roles/${id}`,
+                    this.lightdashConfig.siteUrl,
+                ).href,
+            },
+        };
+    }
+
+    async listRoles({
+        organizationUuid,
+        filter,
+    }: {
+        organizationUuid: string;
+        filter?: string;
+    }): Promise<ScimListResponse<ScimRole>> {
+        this.logger.debug('SCIM: Listing roles', {
+            organizationUuid,
+            filter,
+        });
+        try {
+            const parsedFilter = filter ? parse(filter) : null;
+            this.logger.debug('SCIM: Parsed role filter', { parsedFilter });
+
+            const systemRoles = getSystemRoles();
+
+            // Convert org roles to SCIM format
+            const scimRoles = systemRoles.map((role) =>
+                this.convertLightdashRoleToScimRole(role, ScimRoleType.ORG),
+            );
+
+            // Apply filter if specified
+            let filteredRoles = scimRoles;
+            if (parsedFilter?.op === 'eq') {
+                const { attrPath, compValue } = parsedFilter;
+                filteredRoles = scimRoles.filter((role) => {
+                    switch (attrPath) {
+                        case 'value':
+                            return role.value === compValue;
+                        case 'display':
+                            return role.display === compValue;
+                        case 'type':
+                            return role.type === compValue;
+                        default:
+                            return true;
+                    }
+                });
+            }
+
+            this.logger.debug('SCIM: Successfully listed roles', {
+                organizationUuid,
+                totalResults: filteredRoles.length,
+            });
+
+            return {
```

Comment:
Would it be worth making `ScimListResult` a class to make the response easier to form? I know TS will flag unknown fields, but the DX might improve from having a shared function to create these objects if they're all standard.